haystack/document_store/weaviate.py

```diff
@@ -0,0 +1,592 @@
+import logging
+from typing import Any, Dict, Generator, List, Optional, Union
+import numpy as np
+from tqdm import tqdm
+
+from haystack import Document
+from haystack.document_store.base import BaseDocumentStore
+from haystack.utils import get_batches_from_generator
+
+from weaviate import client, auth, AuthClientPassword
+from weaviate import ObjectsBatchRequest
+
+logger = logging.getLogger(__name__)
+
+class WeaviateDocumentStore(BaseDocumentStore):
+    """
+    https://www.semi.technology/developers/weaviate/current/index.html#what-is-weaviate
+    Weaviate is a cloud-native, modular, real-time vector search engine built to scale your machine learning models.
+    1. Highly efficient large scale vector searches
+    2. State of the Art classification techniques
+    3. Easily retrieve your data in graph format using GraphQL
+
+    Weaviate is a document store to store, query the documents for search
+    Both the document and embedding (generated by Weaviate) are stored in weaviate.
+
+    Weaviate python client is used to connect to the server, more details are here
+    https://weaviate-python-client.readthedocs.io/en/docs/weaviate.html
+
+    Usage:
+    1. Start a Weaviate server (see https://www.semi.technology/developers/weaviate/current/getting-started/installation.html)
+    2. Init a WeaviateDocumentStore in Haystack
+    """
+
+    def __init__(
+            self,
+            weaviate_url: str = "http://localhost:8080",
```

Comment:
Let's keep naming of args as similar to other document stores as possible. 
In this case, I would suggest keeping it similar to elasticsearch and split it into host + port.
(I know we use `sql_url` in FAISS - reasoning back then was that just host confuses people as they would probably look for a FAISS host in a FAISSDocumentstore)